File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        List<Integer> repL = new ArrayList<>();
        HashMap<Integer , Integer> dictN = new HashMap<>();

        for(int num1: nums1){
            dictN.put(num1, dictN.getOrDefault(num1, 0) + 1);
        }
        for(int num2: nums2){
            if(dictN.containsKey(num2) && dictN.get(num2) > 0){
                repL.add(num2);
                dictN.put(num2, dictN.get(num2)-1);
            }
        }
        int [] resArray = new int[repL.size()];
        for(int i=0;i<repL.size();i++){
            resArray[i] = repL.get(i);
        }
        return resArray;

        
    }
}

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.py
class Solution(object):
    def intersect(self, nums1, nums2):
        repL = []
        dictN = {}
        for num1 in nums1:
            if not num1 in dictN:
                dictN[num1] = 1
            else:
                val = 1
                dictN[num1] += val
        print(dictN)
        for num2 in nums2:
            if num2 in dictN and dictN[num2] > 0:
                repL.append(num2)
                dictN[num2] -= 1
        return repL
            


        
        